### PR TITLE
Move input fields to right in segmented buttons

### PR DESCRIPTION
**Pull request by MewenLeHo**
Closes [#1365 ](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/1365)

### CODE REVIEW

**Comment by julien-deramond** [comment]
Not specified in the issue but do we need to do the same (for consistency) for:
* https://deploy-preview-1413--boosted.netlify.app/docs/5.2/forms/input-group/#button-addons
* https://deploy-preview-1413--boosted.netlify.app/docs/5.2/forms/input-group/#custom-select
* https://deploy-preview-1413--boosted.netlify.app/docs/5.2/forms/input-group/#custom-file-input

**Comment by isabellechanclou** [comment]
> Not specified in the issue but do we need to do the same (for consistency) for:

> https://deploy-preview-1413--boosted.netlify.app/docs/5.2/forms/input-group/#button-addons

--> No need to do the same. A callout message is planned to say its not ODS compatible (see #1199)

> https://deploy-preview-1413--boosted.netlify.app/docs/5.2/forms/input-group/#custom-select

--> No need to do the same. A callout message is planned to say its not ODS compatible (see #1199)

> https://deploy-preview-1413--boosted.netlify.app/docs/5.2/forms/input-group/#custom-file-input

--> This part will be redesign by Orange designers and will have to be modified by developers when specifications available.
Moreover, a callout message is planned (see #1199).

**Comment by louismaximepiton** [comment]
Is this PR still needed regarding the fix done on main by Bootstrap ?

**Comment by julien-deramond** [comment]
> Is this PR still needed regarding the fix done on main by Bootstrap ?

What's the status here?

**Comment by isabellechanclou** [comment]
I've just looked at Bootstrap main branch, and the fix described here is still needed, because, not done in Bootstrap.

**Comment by MewenLeHo** [comment]
> I've just looked at Bootstrap main branch, and the fix described here is still needed, because, not done in Bootstrap.

Thanks. Just done the rebase on this old pull request. We are now ready for review...

**Review by louismaximepiton — CHANGES_REQUESTED** [review]
> > Not specified in the issue but do we need to do the same (for consistency) for:
> 
> > https://deploy-preview-1413--boosted.netlify.app/docs/5.2/forms/input-group/#button-addons
> 
> --> No need to do the same. A callout message is planned to say its not ODS compatible (see #1199)
> 
> > https://deploy-preview-1413--boosted.netlify.app/docs/5.2/forms/input-group/#custom-select
> 
> --> No need to do the same. A callout message is planned to say its not ODS compatible (see #1199)
> 
> > https://deploy-preview-1413--boosted.netlify.app/docs/5.2/forms/input-group/#custom-file-input
> 
> --> This part will be redesign by Orange designers and will have to be modified by developers when specifications available. Moreover, a callout message is planned (see #1199).

We need to have consistency over the documentation, so if we do it once, IMO we shall do the changes everywhere.

That said, if we do the changes everywhere, I'm wondering if we can't introduce some rules to make it automated and to be transparent for users. What I have in mind: `.input-group input, .input-group input + * {margin-left: 0;}` or something similar.

File: scss/forms/_input-group.scss [hunk 1]
```diff
@@ -112,11 +112,16 @@
     $validation-messages: $validation-messages + ":not(." + unquote($state) + "-tooltip)" + ":not(." + unquote($state) + "-feedback)";
   }
 
+  // Boosted mod: better display when input or select is invalid
   > :not(:first-child):not(.dropdown-menu)#{$validation-messages} {
-    margin-left: -$input-border-width;
     @include border-start-radius(0);
   }
 
+  > :not(:first-child):not(.dropdown-menu):not(input):not(select):not(input + *):not(select + *)#{$validation-messages} {
```
Comment (hannahiss) [inline scss/forms/_input-group.scss:120]: missing :not(textarea):not(textarea+ *)

**Review by hannahiss — CHANGES_REQUESTED** [review]
Need to add condition for textarea, but otherwise it looks good, in the button addons, segmented buttons, custom select and custom fil input, and also with validation messages in input group.

File: scss/forms/_input-group.scss (hunk 1 above)
Comment (MewenLeHo) [reply scss/forms/_input-group.scss:120]: But this case does not exist?

**Review by hannahiss — APPROVED** [review]
LGTM :-)

**Review by louismaximepiton — APPROVED** [review]
LGTM

**Comment by julien-deramond** [comment]
Are you sure about these renderings?

Screenshot 2022-12-12 at 07 14 14

Screenshot 2022-12-12 at 07 15 02

They seem off compared to the input fields

Screenshot 2022-12-12 at 07 15 42

**Comment by julien-deramond** [comment]
@MewenLeHo Based on the discussion we had with the designers, is this PR still needed?

**Comment by MewenLeHo** [comment]
> @MewenLeHo Based on the discussion we had with the designers, is this PR still needed?

Sadly, no.

**Pull request closed without merging**